src/main/java/org/verdictdb/coordinator/ExecutionContext.java

```diff
@@ -75,20 +85,44 @@ public VerdictResultStream streamsql(String query) throws VerdictDBException {
       SelectQueryCoordinator coordinator =
           new SelectQueryCoordinator(context.getCopiedConnection());
       ExecutionResultReader reader = coordinator.process(query);
-      VerdictResultStream stream = new VerdictResultStream(reader, this);
+      VerdictResultStream stream = new VerdictResultStreamFromExecutionResultReader(reader, this);
       return stream;
     } else if (queryType.equals(QueryType.scrambling)) {
       ScramblingCoordinator coordinator = new ScramblingCoordinator(context.getCopiedConnection());
       return null;
     } else if (queryType.equals(QueryType.set_default_schema)) {
       updateDefaultSchemaFromQuery(query);
       return null;
-
+    } else if (queryType.equals(QueryType.show_databases)) {
+      return generateShowSchemaResultFromQuery();
```

Comment:
Can you assign the return value to a variable; then return the variable? I have found it is easier to debug later.

Please apply the similar approach to the other lines below.